### PR TITLE
renaming bool, wbool/sbool, if/s, etc.

### PR DESCRIPTION
**Pull request by jonsterling**
To avoid ambiguity, the recursor for wbool is now called wbool-rec, and
the recursor for strict bool is just if.

### CODE REVIEW

**Comment by jonsterling** [comment]
We could also just change the parser so that it can recognize which `if` to use, depending on how many arguments it has.

**Comment by favonia** [comment]
How about `wif`?

File: src/redprl/redprl.lex [hunk 1]
```diff
@@ -76,12 +76,12 @@ whitespace = [\ \t];
 
 "?"                => (Tokens.QUESTION (posTuple (size yytext)));
 "fcom"             => (Tokens.FCOM (posTuple (size yytext)));
-"bool"             => (Tokens.BOOL (posTuple (size yytext)));
+"wbool"            => (Tokens.WBOOL (posTuple (size yytext)));
```
Comment (favonia) [inline src/redprl/redprl.lex]: And
```diff
+"wool"             => (Tokens.WBOOL (posTuple (size yytext)));
```

**Comment by jonsterling** [comment]
@favonia I've updated this PR to include `wif` and `wool` 😜

**Comment by favonia** [comment]
@jonsterling Let's hope RedPRL gets popular and then some dedicated user writes a blog post about this. :stuck_out_tongue_closed_eyes:

**Pull request merged**